tensorflow_addons/seq2seq/decoder.py

```diff
@@ -299,14 +300,8 @@ def dynamic_decode(
       `(final_outputs, final_state, final_sequence_lengths)`.
 
     Raises:
-      TypeError: if `decoder` is not an instance of `Decoder`.
       ValueError: if `maximum_iterations` is provided but is not a scalar.
     """
-    if not isinstance(decoder, (Decoder, BaseDecoder)):
```

Comment:
Thanks @guillaumekln, LGTM. May I know why do you remove this?

Reply:
I added the `typechecked` annotation instead.

Reply:
Yeah. Should we also modify `ValueError: ... not a scalar` as `scalar or tensor`?